project_task_code/models/project_task.py

```diff
@@ -28,3 +29,46 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        args = args or []
+        code = self._name_search_parse_code(name)
+        if not code:
+            return super(ProjectTask, self).name_search(
+                name=name,
+                args=args,
+                operator=operator,
+                limit=limit)
+        # We are searching for codes
+        tasks = self.search(
+            [('code', '=ilike', '%' + code + '%')] + args,
```

Comment:
are you sure you are constructing the domain correctly? 
 [('code', '=ilike',  code)] is OK. 

But it is case-insensitive. you may want to make it case insensitive with 'like' in case someone writes the prefix in the wrong case. (if you want to do that you must also change _name_parse_code to support that......)